src/System.Private.CoreLib/shared/System/MemoryExtensions.cs

```diff
@@ -22,6 +22,206 @@ namespace System
     /// </summary>
     public static partial class MemoryExtensions
     {
+        /// <summary>
+        /// Removes all leading and trailing occurrences of a specified element.
+        /// </summary>
+        /// <param name="trimElement">The specified element to look for and remove.</param>
+        public static ReadOnlySpan<T> Trim<T>(this ReadOnlySpan<T> span, T trimElement)
+            where T : IEquatable<T>
+            => TrimHelper(ref span, trimElement, true, true);
+
+        /// <summary>
+        /// Removes all leading occurrences of a specified element.
+        /// </summary>
+        /// <param name="trimElement">The specified element to look for and remove.</param>
+        public static ReadOnlySpan<T> TrimStart<T>(this ReadOnlySpan<T> span, T trimElement)
+            where T : IEquatable<T>
+            => TrimHelper(ref span, trimElement, true, false);
+
+        /// <summary>
+        /// Removes all trailing occurrences of a specified element.
+        /// </summary>
+        /// <param name="trimElement">The specified element to look for and remove.</param>
+        public static ReadOnlySpan<T> TrimEnd<T>(this ReadOnlySpan<T> span, T trimElement)
+            where T : IEquatable<T>
+            => TrimHelper(ref span, trimElement, false, true);
+
+        /// <summary>
+        /// Conditionally removes all leading and trailing occurrences of a specified element.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="span">The source span from which the element is removed.</param>
+        /// <param name="trimElement">The specified element to look for and remove.</param>
+        /// <param name="trimStart">If set to True, trims the start.</param>
+        /// <param name="trimEnd">If set to True, trims the end.</param>
+        private static ReadOnlySpan<T> TrimHelper<T>(ref ReadOnlySpan<T> span, in T trimElement, bool trimStart, bool trimEnd)
```

Comment:
Can the first argument be `in` ?

Reply:
Done - also in the other helper

Reply:
Does passing it as ref/in improve throughput here? On Linux as well?

Reply:
Benched. 
It looks like using the default passing style is superior (on Windows):
```csharp
|        Method |      Mean |     Error |    StdDev | Ratio | RatioSD |
|-------------- |----------:|----------:|----------:|------:|--------:|
| SingleViaNone | 0.0002 ns | 0.0000 ns | 0.0000 ns |  1.00 |    0.00 | // <-- No ref/in
|   SingleViaIn | 0.0002 ns | 0.0000 ns | 0.0000 ns |  1.16 |    0.01 |
|  SingleViaRef | 0.0002 ns | 0.0000 ns | 0.0000 ns |  1.19 |    0.04 |
```